libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/RoadObject.kt

```diff
@@ -1,73 +1,32 @@
 package com.mapbox.navigation.base.trip.model.roadobject
 
 /**
- * Abstract class that serves as a base for all road objects.
+ * Interface that serves as a base for all road objects.
  *
- * Available road object types are:
- * - [TunnelEntrance]
- * - [TunnelExit]
- * - [CountryBorderCrossing]
- * - [TollCollection]
- * - [RestStop]
- * - [RestrictedAreaEntrance]
- * - [RestrictedAreaExit]
- * - [BridgeEntrance]
- * - [BridgeExit]
- * - [Incident]
- * - [Custom]
- *
- * @param objectType constant describing the object type, see [RoadObjectType].
- * @param distanceFromStartOfRoute distance to this object since the start of the route.
- * Will be null for road objects returned by Electronic Horizon.
- * @param objectGeometry geometry details of the object.
  */
-abstract class RoadObject(
-    val objectType: Int,
-    distanceFromStartOfRoute: Double?,
-    val objectGeometry: RoadObjectGeometry,
-) {
+interface RoadObject {
```

Comment:
This interface documentation would very much benefit from a graph that explains the types of objects but more importantly where different types can come from.

For example, that the `RouteAlertLocation` will be produced only for objects that are on the current route that we are actively navigating on, and that those same objects will be duplicated in the e-horizon interface under different types, how is geometry generated, when and why some types are nullable and how do you access the data.

Reply:
Then, we should link to this description from `registerEHorizonObserver`, `registerRoadObjectsObserver`, `routeProgress#upcomingRoadObject` and other places that interact with this structure.

Reply:
Also, should `registerRoadObjectsObserver` be renamed to `registerRoadObjectsOnRouteObserver`? Otherwise, it can be consufed with e-horizon's observer results.

Reply:
> where different types can come from

each object might come from both `directions` and `EH`

`RouteAlertLocation ` is a subtype of `RoadObjectLocation`, not `RoadObject`

Reply:
> that those same objects will be duplicated in the e-horizon interface under different types

the type of the object will be the same. `RoadObjectLocation` will be different

Reply:
> Also, should registerRoadObjectsObserver be renamed to registerRoadObjectsOnRouteObserver? Otherwise, it can be consufed with e-horizon's observer results.

I think no, we've been providing `UpcomingRoadObjects` for both route and EH  for some month, no one customer was confused with it. Also I've added docs to `UpcomingRoadObjects` to clarify some details

Reply:
done